proposals/0303-swiftpm-extensible-build-tools.md

```diff
@@ -61,16 +63,15 @@ To allow plugins to be declared, the following API will be added to `PackageDesc
 extension Target {
 
     /// Defines a new package plugin target with a given name, declaring it as
-    /// providing a capability of adding custom build commands.
+    /// providing a capability of extending SwiftPM in a particular way.
     ///
-    /// The capability determines what kind of build commands it can add. Besides
-    /// determining at what point in the build those commands run, the capability
-    /// determines the context that is available to the plugin and the kinds of
-    /// commands it can create.
+    /// The capability determines the way in which the plugin extends SwiftPM,
+    /// which determines the context that is available to the plugin and the
+    /// kinds of commands it can create.
     ///
-    /// In the initial version of this proposal, three capabilities are provided:
-    /// prebuild, build tool, and postbuild. See the declaration of each capability
-    /// under `PluginCapability` for more information.
+    /// In the initial version of this proposal, only a single build tool plugin
+    /// capability is defined. The intent is to define new capabilities in the
+    /// future.
```

Comment:
since several reviewers pointed out the utility in a capability for tools that run after the builds + test are all complete, I think we can mention that as an one example, and add other well-known examples

Reply:
Yes, that's a great point.  Will update.

Reply:
+1, let's please not remove this. Tasks like "generate documentation" and "upload artifacts" are a perfect example of a post build task.

Reply:
I agree that there are use cases for this.  The problem is that to actually implement those postbuild tasks in a meaningful way would require more information from SwiftPM and much better definitions about exactly when those commands run.  Should they for example run before or after running tests, and how do they get the results of the tests.  For the documentation generation they would often also either need more information from the compiler invocation or enough information to run the compiler themselves.  So I think it would be better to have a follow-on proposal that extends the TargetBuildContext and CommandConstructor APIs (might not even need to be a new task) specifically for these kinds of uses.

Reply:
Hmm... I see, I had hoped to have post build in the next release, so I don't know if new proposal means it's next-next release or not? 🤔

Reply:
> The simplest "postbuild" meaning is "after absolutely everything if nothing failed", so after tests etc. Would have no input about tests etc... The things like "take input from previous task" sound to me like the task DAG which seems far away?

this is also the minimal capability I had in mind, and that I think we should support to begin with

Reply:
I see.  My understanding of the core team feedback was to suggest removing postbuild.  I can put that back, though we'd need to consider what parts of the file system it would make sense for it to have write access to.  Would it for example be able to make any modifications to the built artifacts?

Reply:
I see also from the response to the summary of the first review that there were requests to keep it.  It definitely makes sense, but will require some more definition of semantics here.  I'll add it back.

Reply:
I have added back "postbuild" in the minimal capacity, and added a few paragraphs to the Future Directions about adding a more structured way for SwiftPM to convey information to such a postbuild command in the future.